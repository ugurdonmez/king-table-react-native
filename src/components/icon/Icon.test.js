import React from 'react';
import 'react-native';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Icon from './Icon'

beforeEach(() => {
    isConsoleWarningOrError = false;
    const originalError = global.console.error;
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
        isConsoleWarningOrError = true;
        originalError(...args);
    });
    const originalWarn = global.console.warn;
    jest.spyOn(global.console, 'warn').mockImplementation((...args) => {
        isConsoleWarningOrError = true;
        originalWarn(...args);
    });
});

afterEach(() => {

    if (isConsoleWarningOrError) {
        throw new Error('Console warnings and errors are not allowed');
    }
});

describe('<Icon />', () => {
    it('renders without crashing', () => {
        shallow(<Icon name="Triangle"/>)
    })

    it('Triangle match snapshot', () => {
        const tree = renderer.create(
            <Icon name="Triangle" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('TriangleEmpty match snapshot', () => {
        const tree = renderer.create(
            <Icon name="TriangleEmpty" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('Circle match snapshot', () => {
        const tree = renderer.create(
            <Icon name="Circle" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('CircleEmpty match snapshot', () => {
        const tree = renderer.create(
            <Icon name="CircleEmpty" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})
