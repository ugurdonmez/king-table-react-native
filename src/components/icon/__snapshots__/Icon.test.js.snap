// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> test heigh and width 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 15,
        "width": 15,
      },
    ]
  }
  vbHeight={60.001}
  vbWidth={60.001}
>
  <RNSVGPath
    d="M59.895,58.531l-29-58c-0.34-0.678-1.449-0.678-1.789,0l-29,58c-0.155,0.31-0.139,0.678,0.044,0.973 C0.332,59.798,0.654,59.978,1,59.978h58c0.347,0,0.668-0.18,0.851-0.474C60.033,59.209,60.05,58.841,59.895,58.531z M2.618,57.978 L30,3.214l27.382,54.764H2.618z"
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={0}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
        "fill",
        "fillRule",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  />
</RNSVGSvgView>
`;
